### ISSUE TITLE
Rate limit plugin

### ISSUE THREAD

**Issue by yurirnascimento** [Sammyjo20/Saloon#19]
Hi,

Another need of mine is to be able to use a request rate limit, because normally my jobs run for long periods but need to respect the API's request limit.

I'm creating a plugin to use the [Spatie/GuzzleRateLimiterMIddleware](https://packagist.org/packages/spatie/guzzle-rate-limiter-middleware)

Would it be useful to anyone else?

**Comment by Sammyjo20** [comment]
Yes, this would be great! I wonder if community made plugins could be released as standalone packages. Is this something you would be up for building?

**Comment by yurirnascimento** [comment]
This would be a useful plugin for me, so I could mantain it, what do you think?

**Comment by Sammyjo20** [comment]
If you are happy to, I would add it to a list of community driven plugins for Saloon.

**Comment by Sammyjo20** [comment]
Just a suggestion, it would be cool if the plugin also added two methods to "set" and "get" from a cache, so it would be really easy for Laravel users to hook it up to their Laravel app. 

![Screen Shot 2022-01-28 at 21 42 13](https://user-images.githubusercontent.com/29132017/151625581-ff0d5e62-2af2-47e9-b47c-7926018571c3.png)

**Comment by yurirnascimento** [comment]
I'll probably do something simple, and expose the API for the user to customize as needed.

**Comment by Sammyjo20** [comment]
Sounds good. Looking forward to seeing it.

**Comment by Sammyjo20** [comment]
Just a thought, I need a rate limiter in Laravel, but I'd probably prefer if it threw an exception that I can catch in my job and delay it by X amount of seconds that the API replies with. Maybe that's something to consider in another plugin? Like a back-off plugin that uses Laravel Cache to determine if a rate limit has been exceeded

**Comment by yurirnascimento** [comment]
Hi, I'm still on it, I'll be back soon with news.
